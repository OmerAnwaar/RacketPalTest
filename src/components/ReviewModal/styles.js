import {StyleSheet} from 'react-native';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen';
import {colors, theme} from '../../shared/constants';

const styles = StyleSheet.create({
  container: {
    ...theme.SINGLE_FLEX,
    ...theme.CENTER,
    backgroundColor: colors.TRANSPERANT,
  },
  modalContainer: {
    height: 439,
    borderRadius: 12,
    backgroundColor: colors.WHITE,
  },

  containerWidth: {width: '90%'},
  title: {
    fontWeight: '700',
    fontSize: 16,
    textAlign: 'center',
  },
  subTitle: {
    fontSize: 14,
    lineHeight: 16.8,
    fontWeight: '400',
    textAlign: 'center',
    paddingHorizontal: 40,
  },
  feedbackText: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: '400',
    textAlign: 'center',
    paddingHorizontal: 40,
    textDecorationLine: 'underline',
  },
  iconStyle: {
    ...theme.ICON_DIMEN,
  },
  iconStyleContainer: {
    alignSelf: 'flex-end',
    marginEnd: wp(4),
  },
  reviewStars: {
    height: 80,
    width: '100%',
  },
  subContainer: {
    ...theme.SINGLE_FLEX,
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingHorizontal: '5%',
  },

  //STAR MODAL

  starContainer: {
    height: 242,
    borderRadius: 12,
    backgroundColor: colors.WHITE,
  },
  editIcon: {
    height: 47,
    width: 47,
  },
  starTitle: {
    fontSize: 16,
    paddingTop: '15%',
    fontWeight: '700',
    textAlign: 'center',
  },
  editContainer: {
    top: -20,
    width: 84,
    height: 84,
    ...theme.CENTER,
    borderRadius: 45,
    position: 'absolute',
    backgroundColor: colors.WHITE,
  },
  subTitleStar: {
    fontSize: 12,
    lineHeight: 16.8,
    fontWeight: '400',
    textAlign: 'center',
    paddingHorizontal: 40,
  },
  label: {
    fontSize: 10,
    lineHeight: 15,
    fontWeight: '700',
    textAlign: 'left',
    color: colors.PRIMARY,
  },
  remindmelater: {
    fontSize: 10,
    lineHeight: 15,
    fontWeight: '700',
    textAlign: 'center',
    paddingHorizontal: 40,
    textTransform: 'uppercase',
    color: colors.NOBEL,
  },
  starSubContainer: {
    paddingVertical: 10,
    width: 300,
    justifyContent: 'space-evenly',
  },
  inputStyle: {
    width: '100%',
    borderWidth: 1,
    borderColor: colors.PRIMARY,
    borderRadius: 6,
    height: 112,
    padding: '3%',
    paddingTop: '3%',
  },
  submitButton: {width: '100%', backgroundColor: colors.PRIMARY},
});

export default styles;
