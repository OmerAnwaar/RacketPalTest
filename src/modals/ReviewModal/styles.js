import {StyleSheet} from 'react-native';
import {colors, theme} from '../../shared/constants';

const styles = StyleSheet.create({
  container: {
    ...theme.SINGLE_FLEX,
    ...theme.CENTER,
    backgroundColor: colors.TRANSPERANT,
  },
});

export default styles;
