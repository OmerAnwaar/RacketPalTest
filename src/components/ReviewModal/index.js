import {useNavigation} from '@react-navigation/native';
import React, {useState} from 'react';
import {
  Image,
  Modal,
  KeyboardAvoidingView,
  Pressable,
  Text,
  TextInput,
  View,
  Platform,
  Keyboard,
  Linking,
} from 'react-native';
import {AirbnbRating} from 'react-native-ratings';
import {colors, icons} from '../../shared/constants';
import RpButton from '../RpButton';
import styles from './styles';

const appleStore = 'itms-apps://itunes.apple.com/us/app/id1453817491?mt=8';
const googlePlayStore = 'market://details?id=com.racketpal';

export default function ReviewModal({data, visible, handlePress, setVisible}) {
  const [star, setStar] = useState(false);
  const [rating, setRating] = useState(0);
  const navigation = useNavigation();

  function handleCancel() {
    setStar(false);
    setVisible(false);
    setRating(0);
  }
  function ratingCompleted(rating) {
    setRating(rating);
    if (rating > 3) {
      handleCancel();
      Linking.openURL(Platform.OS === 'ios' ? appleStore : googlePlayStore);
    }
  }

  return (
    <Modal visible={visible} transparent animationType={'slide'}>
      <Pressable style={styles.container}>
        {!star ? (
          <View style={styles.containerWidth}>
            <View style={styles.modalContainer}>
              <View style={styles.subContainer}>
                <Pressable
                  onPress={() => setVisible(false)}
                  style={styles.iconStyleContainer}>
                  <Image source={icons.CLOSE} style={styles.iconStyle} />
                </Pressable>
                <Image
                  source={icons.REVIEW_STARS}
                  style={styles.reviewStars}
                  resizeMode={'contain'}
                />
                <Text style={styles.title}>Enjoying RacketPal?</Text>
                <Text style={styles.subTitle}>
                  Your App Store review greatly helps spread the word and grow
                  the racket sports community!
                </Text>
                <RpButton
                  label={'Rate us'}
                  onPress={() => {
                    setStar(true);
                  }}
                />
                <Pressable
                  onPress={() => {
                    setVisible(false);
                    navigation.navigate('Contact Us');
                  }}>
                  <Text style={styles.feedbackText}>
                    Not yet? Give us feedback
                  </Text>
                </Pressable>
              </View>
            </View>
          </View>
        ) : (
          <KeyboardAvoidingView
            behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
            <Pressable
              onPress={() => Keyboard.dismiss()}
              style={styles.containerWidth}>
              <View
                style={
                  rating
                    ? {...styles.starContainer, height: 400}
                    : styles.starContainer
                }>
                <View style={styles.subContainer}>
                  <View style={styles.editContainer}>
                    <Image
                      source={icons.EDIT}
                      style={styles.editIcon}
                      resizeMode={'contain'}
                    />
                  </View>
                  <Text style={styles.starTitle}>Enjoying RacketPal?</Text>
                  <Text style={styles.subTitleStar}>
                    Tap a star to rate it on the App store
                  </Text>
                  <AirbnbRating
                    size={30}
                    count={5}
                    showRating={false}
                    defaultRating={0}
                    selectedColor={colors.PRIMARY}
                    onFinishRating={ratingCompleted}
                    starContainerStyle={styles.starSubContainer}

                    // starImage={icons.STAR}
                  />
                  {rating < 4 && rating != 0 ? (
                    <>
                      <Text style={styles.label}>ANY FEEDBACK FOR US?</Text>
                      <TextInput
                        style={styles.inputStyle}
                        selectionColor={colors.PRIMARY}
                        multiline={true}
                      />
                      <RpButton
                        label={'Submit'}
                        style={styles.submitButton}
                        onPress={handleCancel}
                      />
                    </>
                  ) : (
                    <Pressable onPress={handleCancel}>
                      <Text style={styles.remindmelater}>remind me later</Text>
                    </Pressable>
                  )}
                </View>
              </View>
            </Pressable>
          </KeyboardAvoidingView>
        )}
      </Pressable>
    </Modal>
  );
}
