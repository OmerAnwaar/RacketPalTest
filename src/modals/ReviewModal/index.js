import {useNavigation} from '@react-navigation/native';
import RpRateUsCard from '../../components/Card/RpRateusCard';
import RpRatingCard from '../../components/Card/RpRatingCard';

import React, {useState} from 'react';
import {Keyboard, Linking, Modal, Platform, Pressable} from 'react-native';

import styles from './styles';
import {logAnalytic} from '../../shared/utils';

const appleStore = 'itms-apps://itunes.apple.com/us/app/id1453817491?mt=8';
const googlePlayStore = 'market://details?id=com.racketpal';

const ReviewModal = ({visible, toggleModal}) => {
  const navigation = useNavigation();

  const [preRating, setPreRating] = useState(false);
  const [rating, setRating] = useState(0);

  function handleCancel() {
    logAnalytic('submit', '123');
    setPreRating(false);
    toggleModal();
    setRating(0);
  }

  function ratingCompleted(userRating) {
    setRating(userRating);
    if (userRating > 3) {
      handleCancel();
      Linking.openURL(Platform.OS === 'ios' ? appleStore : googlePlayStore);
    }
  }

  function handleDismiss() {
    logAnalytic('Dismissed Keyboard', '123');
    Keyboard.dismiss();
  }

  function handleRateUs() {
    logAnalytic('RateUs', '1234');
    setPreRating(true);
  }

  function handleFeedback() {
    logAnalytic('Not yet', '351312');
    toggleModal();
    navigation.navigate('Contact Us');
  }

  return (
    <Modal visible={visible} transparent animationType={'slide'}>
      <Pressable style={styles.container}>
        {!preRating ? (
          <RpRateUsCard
            testId={'feedback'}
            onRateUs={handleRateUs}
            onFeedback={handleFeedback}
            handleVisible={toggleModal}
          />
        ) : (
          <RpRatingCard
            onRatingCompleted={ratingCompleted}
            rating={rating}
            onRemindMeLater={handleCancel}
            onSubmit={handleCancel}
            onDismiss={handleDismiss}
          />
        )}
      </Pressable>
    </Modal>
  );
};
export default ReviewModal;
