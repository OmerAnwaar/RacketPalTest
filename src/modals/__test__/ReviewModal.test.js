import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import ReviewModal from '../ReviewModal';

const mockedFirebaseAnalyticsLogEvent = jest.fn();
const mockedFirebaseAnalyticsLogSelectContent = jest.fn();

jest.mock('@react-native-firebase/analytics', () => () => {
  return {
    logEvent: mockedFirebaseAnalyticsLogEvent,
    logSelectContent: mockedFirebaseAnalyticsLogSelectContent,
  };
});

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('Review Component Screen', () => {
  it('renders Review Modal correctly', () => {
    const rendered = render(
      <ReviewModal visible={'false'} toggleModal={jest.fn()} />,
    );
    expect(rendered).toMatchSnapshot();
  });

  // it('should render handle feedback', () => {
  //   const testIdName = 'feedback';
  //   const {findByTestId} = render(
  //     <ReviewModal visible={'false'} toggleModal={jest.fn()} />,
  //   );
  //   fireEvent.press(findByTestId(testIdName));
  // });
});
