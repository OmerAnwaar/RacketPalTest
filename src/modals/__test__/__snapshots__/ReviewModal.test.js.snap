// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component Screen renders Review Modal correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible="false"
/>
`;
